packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesUtils.kt

```diff
@@ -139,7 +139,10 @@ object UpdatesUtils {
       } catch (e: NoSuchFileException) {
         throw IOException("File download was successful, but temp file ${tmpFile.absolutePath} does not exist")
       } catch (e: FileAlreadyExistsException) {
-        throw IOException("File download was successful, but file already exists at ${destination.absolutePath}")
+        // ENG-10888: this will happen any time there are two assets with identical content
+        // (and therefore the same hash). Log this condition, but do not throw an exception.
+        // throw IOException("File download was successful, but file already exists at ${destination.absolutePath}")
+        Log.w(TAG, "File download was successful, but file already exists at ${destination.absolutePath}")
```

Comment:
do we need to verify that it is the same file contents? (i.e. hashes are the same) Would it make more sense to just overwrite the file if it exists?

Reply:
I guess by the way it is called, destination is `key + extension` where key happens to be a hash most of the time, but I'm not sure it's true in 100% of cases (hard to trace, and even if it were easy to trace this function would be safer in isolation if we didn't assume anything about the input), so we should probably be safe and ensure we're either a) overwriting it, or b) checking that if a file already exists there it is the same file.

Reply:
Fixed